File: framec_tests/java/Naming/Naming.java
// emitted from framec_v0.10.0
// get include files at https://github.com/frame-lang/frame-ancillary-files

package framec_tests.java.Naming;
import java.util.*;
import framec_tests.java.FrameLang.FrameEvent;

class Naming {
    
    private NamingCompartment _compartment_;
    private NamingCompartment _nextCompartment_;
    
    
    
    Naming() {
        
        // Create and intialize start state compartment.
        this._compartment_ = new NamingCompartment(this._state_);
        this._nextCompartment_ = null;
        
        
        // Send system start event
        FrameEvent frame_event = new FrameEvent(">", null);
        this._mux_(frame_event);
    
    }
    
    // states enum
    private enum NamingState {
        
        INIT(0),
        SNAKE_STATE(1),
        CAMELSTATE(2),
        STATE123(3),
        FINAL(4);
        
        public final int value;
        
        private NamingState(int value) {
            this.value=value;
        }
        
        public int getValue() {
            return value;
        }
    }
    
    //====================== Multiplexer ====================//
    
    private void _mux_(FrameEvent e) {
        if(this._compartment_.state == NamingState.INIT.getValue()) {
            this._sInit_(e);
        }else if(this._compartment_.state == NamingState.SNAKE_STATE.getValue()) {
            this._ssnake_state_(e);
        }else if(this._compartment_.state == NamingState.CAMELSTATE.getValue()) {
            this._sCamelState_(e);
        }else if(this._compartment_.state == NamingState.STATE123.getValue()) {
            this._sstate123_(e);
        }else if(this._compartment_.state == NamingState.FINAL.getValue()) {
            this._sFinal_(e);
        }
        
        
        if(this._nextCompartment_ != null) {
            NamingCompartment nextCompartment = this._nextCompartment_;
            this._nextCompartment_ = null;
            if(nextCompartment._forwardEvent != null && 
             nextCompartment._forwardEvent._message == ">") {
                this._mux_(new FrameEvent( "<", this._compartment_.exitArgs));
                this._compartment_ = nextCompartment;
                this._mux_(nextCompartment._forwardEvent);
            } else {
                this._doTransition_(nextCompartment);
                if(nextCompartment._forwardEvent != null) {
                    this._mux_(nextCompartment._forwardEvent);
                }
            }
            nextCompartment._forwardEvent = null;
        }
    }
    
    //===================== Interface Block ===================//
    
    public void snake_event(int snake_param) {
        HashMap<String,Object> parameters = new HashMap<String,Object>();
        parameters.put("snake_param", snake_param);

        FrameEvent e = new FrameEvent("snake_event", parameters);
        this._mux_(e);
    }
    
    public void CamelEvent(int CamelParam) {
        HashMap<String,Object> parameters = new HashMap<String,Object>();
        parameters.put("CamelParam", CamelParam);

        FrameEvent e = new FrameEvent("CamelEvent", parameters);
        this._mux_(e);
    }
    
    public void event123(int param123) {
        HashMap<String,Object> parameters = new HashMap<String,Object>();
        parameters.put("param123", param123);

        FrameEvent e = new FrameEvent("event123", parameters);
        this._mux_(e);
    }
    
    public void call(String event,int param) {
        HashMap<String,Object> parameters = new HashMap<String,Object>();
        parameters.put("event", event);

        parameters.put("param", param);

        FrameEvent e = new FrameEvent("call", parameters);
        this._mux_(e);
    }
    
    
    //===================== Machine Block ===================//
    
    private void _sInit_(FrameEvent e) {
        if(e._message == "snake_event") {
            NamingCompartment compartment =  new NamingCompartment(NamingState.SNAKE_STATE.getValue());
            compartment.stateArgs.put("snake_state_param", (int) e._parameters.get("snake_param"));
            compartment.stateVars.put("snake_state_var", this.snake_domain_var + this.CamelDomainVar + this.domainVar123 + 100);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "CamelEvent") {
            NamingCompartment compartment =  new NamingCompartment(NamingState.CAMELSTATE.getValue());
            compartment.stateArgs.put("CamelStateParam", (int) e._parameters.get("CamelParam"));
            compartment.stateVars.put("CamelStateVar", this.snake_domain_var + this.CamelDomainVar + this.domainVar123 + 200);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "event123") {
            NamingCompartment compartment =  new NamingCompartment(NamingState.STATE123.getValue());
            compartment.stateArgs.put("stateParam123", (int) e._parameters.get("param123"));
            compartment.stateVars.put("stateVar123", this.snake_domain_var + this.CamelDomainVar + this.domainVar123 + 300);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "call") {
            if (((String) e._parameters.get("event")) == "snake_event") {
                snake_event(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "CamelEvent") {
                CamelEvent(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "event123") {
                event123(((int) e._parameters.get("param")));
            } else {
            }
            return;
        }
    }
    
    private void _ssnake_state_(FrameEvent e) {
          //  1100
        if(e._message == "snake_event") {
            int snake_local_var  = (int) this._compartment_.stateVars.get("snake_state_var") + (int) this._compartment_.stateArgs.get("snake_state_param") + (int) e._parameters.get("snake_param");
            snake_action_do(snake_local_var);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", snake_local_var);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "CamelEvent") {
            int CamelLocalVar  = (int) this._compartment_.stateVars.get("snake_state_var") + (int) this._compartment_.stateArgs.get("snake_state_param") + (int) e._parameters.get("CamelParam");
            CamelAction_do(CamelLocalVar);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", CamelLocalVar);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "event123") {
            int localVar123  = (int) this._compartment_.stateVars.get("snake_state_var") + (int) this._compartment_.stateArgs.get("snake_state_param") + (int) e._parameters.get("param123");
            action123_do(localVar123);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", localVar123);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "call") {
            if (((String) e._parameters.get("event")) == "snake_event") {
                snake_event(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "CamelEvent") {
                CamelEvent(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "event123") {
                event123(((int) e._parameters.get("param")));
            } else {
            }
            return;
        }
    }
    
    private void _sCamelState_(FrameEvent e) {
          //  1200
        if(e._message == "snake_event") {
            int snake_local_var  = (int) this._compartment_.stateVars.get("CamelStateVar") + (int) this._compartment_.stateArgs.get("CamelStateParam") + (int) e._parameters.get("snake_param");
            snake_action_do(snake_local_var);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", snake_local_var);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "CamelEvent") {
            int CamelLocalVar  = (int) this._compartment_.stateVars.get("CamelStateVar") + (int) this._compartment_.stateArgs.get("CamelStateParam") + (int) e._parameters.get("CamelParam");
            CamelAction_do(CamelLocalVar);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", CamelLocalVar);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "event123") {
            int localVar123  = (int) this._compartment_.stateVars.get("CamelStateVar") + (int) this._compartment_.stateArgs.get("CamelStateParam") + (int) e._parameters.get("param123");
            action123_do(localVar123);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", localVar123);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "call") {
            if (((String) e._parameters.get("event")) == "snake_event") {
                snake_event(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "CamelEvent") {
                CamelEvent(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "event123") {
                event123(((int) e._parameters.get("param")));
            } else {
            }
            return;
        }
    }
    
    private void _sstate123_(FrameEvent e) {
          //  1300
        if(e._message == "snake_event") {
            int snake_local_var  = (int) this._compartment_.stateVars.get("stateVar123") + (int) this._compartment_.stateArgs.get("stateParam123") + (int) e._parameters.get("snake_param");
            snake_action_do(snake_local_var);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", snake_local_var);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "CamelEvent") {
            int CamelLocalVar  = (int) this._compartment_.stateVars.get("stateVar123") + (int) this._compartment_.stateArgs.get("stateParam123") + (int) e._parameters.get("CamelParam");
            CamelAction_do(CamelLocalVar);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", CamelLocalVar);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "event123") {
            int localVar123  = (int) this._compartment_.stateVars.get("stateVar123") + (int) this._compartment_.stateArgs.get("stateParam123") + (int) e._parameters.get("param123");
            action123_do(localVar123);
            NamingCompartment compartment =  new NamingCompartment(NamingState.FINAL.getValue());
            compartment.stateArgs.put("result", localVar123);
            
            this._transition_(compartment);
            return;
        }
        else if(e._message == "call") {
            if (((String) e._parameters.get("event")) == "snake_event") {
                snake_event(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "CamelEvent") {
                CamelEvent(((int) e._parameters.get("param")));
            } else if (((String) e._parameters.get("event")) == "event123") {
                event123(((int) e._parameters.get("param")));
            } else {
            }
            return;
        }
    }
    
    private void _sFinal_(FrameEvent e) {
        if(e._message == ">") {
            logFinal_do(((int) this._compartment_.stateArgs.get("result")));
            NamingCompartment compartment =  new NamingCompartment(NamingState.INIT.getValue());
            
            this._transition_(compartment);
            return;
        }
    }
    
    //===================== Actions Block ===================//
    
    // Unimplemented Actions
    
    protected void snake_action_do(int snake_param) { throw new UnsupportedOperationException(); }
    protected void CamelAction_do(int CamelParam) { throw new UnsupportedOperationException(); }
    protected void action123_do(int param123) { throw new UnsupportedOperationException(); }
    protected void logFinal_do(int r) { throw new UnsupportedOperationException(); }
    
    //===================== Domain Block ===================//
    
    public int snake_domain_var  = 300;
    public int CamelDomainVar  = 550;
    public int domainVar123  = 150;
    public ArrayList<Integer> snake_log  = new ArrayList<Integer>();
    public ArrayList<Integer> CamelLog  = new ArrayList<Integer>();
    public ArrayList<Integer> log123  = new ArrayList<Integer>();
    public ArrayList<Integer> finalLog  = new ArrayList<Integer>();
    
    //=============== Machinery and Mechanisms ==============//
    
    private int _state_;
    
    private void _transition_(NamingCompartment compartment) {
        this._nextCompartment_ = compartment;
    }
    
    private void _doTransition_(NamingCompartment nextCompartment) {
        this._mux_(new FrameEvent("<", this._compartment_.exitArgs));
        this._compartment_ = nextCompartment;
        this._mux_(new FrameEvent(">", this._compartment_.enterArgs));
    }
    
    public String state_info(){
        return String.valueOf(this._compartment_.state);
            }
            
    }
    
    //=============== Compartment ==============//
    
    class NamingCompartment {
    
        int state;
        
        NamingCompartment(int state) {
            this.state = state;
        }
        
        HashMap<String, Object> stateArgs = new HashMap<String, Object>();
        HashMap<String, Object> stateVars = new HashMap<String, Object>();
        HashMap<String, Object> enterArgs = new HashMap<String, Object>();
        HashMap<String, Object> exitArgs = new HashMap<String, Object>();
        FrameEvent _forwardEvent = new FrameEvent();
    }
    
    
    /********************

    class NamingController extends Naming {

    	NamingController() {
    	  super();
    	}
    
    protected void snake_action_do(int snake_param) {}
    
    protected void CamelAction_do(int CamelParam) {}
    
    protected void action123_do(int param123) {}
    
    protected void logFinal_do(int r) {}
    }
    
********************/
    

File: framec_tests/golang/naming/naming.go
// emitted from framec_v0.11.0
// get include files at https://github.com/frame-lang/frame-ancillary-files


func NewNaming() Naming {
    m := &namingStruct{}
    
    // Validate interfaces
    var _ Naming = m
    var _ Naming_actions = m
    
    // Create and intialize start state compartment.
    m._compartment_ = NewNamingCompartment(NamingState_Init)
    
    // Override domain variables.
    m.snake_domain_var = 300
    m.CamelDomainVar = 550
    m.domainVar123 = 150
    m.snake_log = []int{}
    m.CamelLog = []int{}
    m.log123 = []int{}
    m.finalLog = []int{}
    
    // Send system start event
    e := framelang.FrameEvent{Msg:">"}
    m._mux_(&e)
    return m
}


type NamingState uint

const (
    NamingState_Init NamingState = iota
    NamingState_snake_state
    NamingState_CamelState
    NamingState_state123
    NamingState_Final
)

type Naming interface {
    Snake_event(snake_param int) 
    CamelEvent(CamelParam int) 
    Event123(param123 int) 
    Call(event string,param int) 
}

type Naming_actions interface {
    snake_action(snake_param int) 
    CamelAction(CamelParam int) 
    action123(param123 int) 
    logFinal(r int) 
}


type namingStruct struct {
    _compartment_ *NamingCompartment
    _nextCompartment_ *NamingCompartment
    snake_domain_var int
    CamelDomainVar int
    domainVar123 int
    snake_log []int
    CamelLog []int
    log123 []int
    finalLog []int
}

//===================== Interface Block ===================//

func (m *namingStruct) Snake_event(snake_param int)  {
    params := make(map[string]interface{})
    params["snake_param"] = snake_param
    e := framelang.FrameEvent{Msg:"snake_event", Params:params}
    m._mux_(&e)
}

func (m *namingStruct) CamelEvent(CamelParam int)  {
    params := make(map[string]interface{})
    params["CamelParam"] = CamelParam
    e := framelang.FrameEvent{Msg:"CamelEvent", Params:params}
    m._mux_(&e)
}

func (m *namingStruct) Event123(param123 int)  {
    params := make(map[string]interface{})
    params["param123"] = param123
    e := framelang.FrameEvent{Msg:"event123", Params:params}
    m._mux_(&e)
}

func (m *namingStruct) Call(event string,param int)  {
    params := make(map[string]interface{})
    params["event"] = event
    params["param"] = param
    e := framelang.FrameEvent{Msg:"call", Params:params}
    m._mux_(&e)
}

//====================== Multiplexer ====================//

func (m *namingStruct) _mux_(e *framelang.FrameEvent) {
    switch m._compartment_.State {
    case NamingState_Init:
        m._NamingState_Init_(e)
    case NamingState_snake_state:
        m._NamingState_snake_state_(e)
    case NamingState_CamelState:
        m._NamingState_CamelState_(e)
    case NamingState_state123:
        m._NamingState_state123_(e)
    case NamingState_Final:
        m._NamingState_Final_(e)
    }
    
    if m._nextCompartment_ != nil {
        nextCompartment := m._nextCompartment_
        m._nextCompartment_ = nil
        if nextCompartment._forwardEvent_ != nil && 
           nextCompartment._forwardEvent_.Msg == ">" {
            m._mux_(&framelang.FrameEvent{Msg: "<", Params: m._compartment_.ExitArgs, Ret: nil})
            m._compartment_ = nextCompartment
            m._mux_(nextCompartment._forwardEvent_)
        } else {
            m._do_transition_(nextCompartment)
            if nextCompartment._forwardEvent_ != nil {
                m._mux_(nextCompartment._forwardEvent_)
            }
        }
        nextCompartment._forwardEvent_ = nil
    }
}

//===================== Machine Block ===================//

func (m *namingStruct) _NamingState_Init_(e *framelang.FrameEvent) {
    switch e.Msg {
    case "snake_event":
        compartment := NewNamingCompartment(NamingState_snake_state)
        compartment.StateArgs["snake_state_param"] = e.Params["snake_param"].(int)
        
        compartment.StateVars["snake_state_var"] = m.snake_domain_var + m.CamelDomainVar + m.domainVar123 + 100
        
        m._transition_(compartment)
        
        return
    case "CamelEvent":
        compartment := NewNamingCompartment(NamingState_CamelState)
        compartment.StateArgs["CamelStateParam"] = e.Params["CamelParam"].(int)
        
        compartment.StateVars["CamelStateVar"] = m.snake_domain_var + m.CamelDomainVar + m.domainVar123 + 200
        
        m._transition_(compartment)
        
        return
    case "event123":
        compartment := NewNamingCompartment(NamingState_state123)
        compartment.StateArgs["stateParam123"] = e.Params["param123"].(int)
        
        compartment.StateVars["stateVar123"] = m.snake_domain_var + m.CamelDomainVar + m.domainVar123 + 300
        
        m._transition_(compartment)
        
        return
    case "call":
        if e.Params["event"].(string) == "snake_event" {
            m.Snake_event(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "CamelEvent" {
            m.CamelEvent(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "event123" {
            m.Event123(e.Params["param"].(int))
            return
        } else {
        }
        
        return
    }
}

func (m *namingStruct) _NamingState_snake_state_(e *framelang.FrameEvent) {
    switch e.Msg {
      //  1100
    case "snake_event":
        var snake_local_var int = m._compartment_.StateVars["snake_state_var"].(int) + m._compartment_.StateArgs["snake_state_param"].(int) + e.Params["snake_param"].(int)
        m.snake_action(snake_local_var)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = snake_local_var
        
        m._transition_(compartment)
        
        return
    case "CamelEvent":
        var CamelLocalVar int = m._compartment_.StateVars["snake_state_var"].(int) + m._compartment_.StateArgs["snake_state_param"].(int) + e.Params["CamelParam"].(int)
        m.CamelAction(CamelLocalVar)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = CamelLocalVar
        
        m._transition_(compartment)
        
        return
    case "event123":
        var localVar123 int = m._compartment_.StateVars["snake_state_var"].(int) + m._compartment_.StateArgs["snake_state_param"].(int) + e.Params["param123"].(int)
        m.action123(localVar123)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = localVar123
        
        m._transition_(compartment)
        
        return
    case "call":
        if e.Params["event"].(string) == "snake_event" {
            m.Snake_event(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "CamelEvent" {
            m.CamelEvent(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "event123" {
            m.Event123(e.Params["param"].(int))
            return
        } else {
        }
        
        return
    }
}

func (m *namingStruct) _NamingState_CamelState_(e *framelang.FrameEvent) {
    switch e.Msg {
      //  1200
    case "snake_event":
        var snake_local_var int = m._compartment_.StateVars["CamelStateVar"].(int) + m._compartment_.StateArgs["CamelStateParam"].(int) + e.Params["snake_param"].(int)
        m.snake_action(snake_local_var)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = snake_local_var
        
        m._transition_(compartment)
        
        return
    case "CamelEvent":
        var CamelLocalVar int = m._compartment_.StateVars["CamelStateVar"].(int) + m._compartment_.StateArgs["CamelStateParam"].(int) + e.Params["CamelParam"].(int)
        m.CamelAction(CamelLocalVar)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = CamelLocalVar
        
        m._transition_(compartment)
        
        return
    case "event123":
        var localVar123 int = m._compartment_.StateVars["CamelStateVar"].(int) + m._compartment_.StateArgs["CamelStateParam"].(int) + e.Params["param123"].(int)
        m.action123(localVar123)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = localVar123
        
        m._transition_(compartment)
        
        return
    case "call":
        if e.Params["event"].(string) == "snake_event" {
            m.Snake_event(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "CamelEvent" {
            m.CamelEvent(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "event123" {
            m.Event123(e.Params["param"].(int))
            return
        } else {
        }
        
        return
    }
}

func (m *namingStruct) _NamingState_state123_(e *framelang.FrameEvent) {
    switch e.Msg {
      //  1300
    case "snake_event":
        var snake_local_var int = m._compartment_.StateVars["stateVar123"].(int) + m._compartment_.StateArgs["stateParam123"].(int) + e.Params["snake_param"].(int)
        m.snake_action(snake_local_var)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = snake_local_var
        
        m._transition_(compartment)
        
        return
    case "CamelEvent":
        var CamelLocalVar int = m._compartment_.StateVars["stateVar123"].(int) + m._compartment_.StateArgs["stateParam123"].(int) + e.Params["CamelParam"].(int)
        m.CamelAction(CamelLocalVar)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = CamelLocalVar
        
        m._transition_(compartment)
        
        return
    case "event123":
        var localVar123 int = m._compartment_.StateVars["stateVar123"].(int) + m._compartment_.StateArgs["stateParam123"].(int) + e.Params["param123"].(int)
        m.action123(localVar123)
        compartment := NewNamingCompartment(NamingState_Final)
        compartment.StateArgs["result"] = localVar123
        
        m._transition_(compartment)
        
        return
    case "call":
        if e.Params["event"].(string) == "snake_event" {
            m.Snake_event(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "CamelEvent" {
            m.CamelEvent(e.Params["param"].(int))
            return
        } else if e.Params["event"].(string) == "event123" {
            m.Event123(e.Params["param"].(int))
            return
        } else {
        }
        
        return
    }
}

func (m *namingStruct) _NamingState_Final_(e *framelang.FrameEvent) {
    switch e.Msg {
    case ">":
        m.logFinal((m._compartment_.StateArgs["result"].(int)))
        compartment := NewNamingCompartment(NamingState_Init)
        m._transition_(compartment)
        
        return
    }
}

//=============== Machinery and Mechanisms ==============//

func (m *namingStruct) _transition_(compartment *NamingCompartment) {
    m._nextCompartment_ = compartment
}

func (m *namingStruct) _do_transition_(nextCompartment *NamingCompartment) {
    m._mux_(&framelang.FrameEvent{Msg: "<", Params: m._compartment_.ExitArgs, Ret: nil})
    m._compartment_ = nextCompartment
    m._mux_(&framelang.FrameEvent{Msg: ">", Params: m._compartment_.EnterArgs, Ret: nil})
}

//===================== Actions Block ===================//


/********************************************************

// Unimplemented Actions

func (m *namingStruct) snake_action(snake_param int)  {}
func (m *namingStruct) CamelAction(CamelParam int)  {}
func (m *namingStruct) action123(param123 int)  {}
func (m *namingStruct) logFinal(r int)  {}

********************************************************/

//=============== Compartment ==============//

type NamingCompartment struct {
    State NamingState
    StateArgs map[string]interface{}
    StateVars map[string]interface{}
    EnterArgs map[string]interface{}
    ExitArgs map[string]interface{}
    _forwardEvent_ *framelang.FrameEvent
}

func NewNamingCompartment(state NamingState) *NamingCompartment {
    c := &NamingCompartment{State: state}
    c.StateArgs = make(map[string]interface{})
    c.StateVars = make(map[string]interface{})
    c.EnterArgs = make(map[string]interface{})
    c.ExitArgs = make(map[string]interface{})
    return c
}